server/src/main/scala/services/graphstages/ArrivalsGraphStage.scala

```diff
@@ -260,23 +260,93 @@ class ArrivalsGraphStage(name: String = "",
       mergeBestFieldsFromSources(baseArrival, mergeArrival(key).getOrElse(baseArrival))
     }
 
+
+    def scheduleTimeApproximation(searchKey: UniqueArrival, searchKeyArrivalsSourceType: ArrivalsSourceType, foundArrival: Arrival, arrivalsSourceType: ArrivalsSourceType, logMessage: Boolean = false): Option[UniqueArrival] = {
```

Comment:
A few suggestions for this function:
- drop the `logMessage` boolean. we can control the logging with the log level - use debug if it's not something we generally want to be logged
- we're only using the `Origin` field on `foundArrival` so we only need to pass that value
- `searchKeyArrivalsSourceType` could be shortened to `searchSourceType` to make it more readable